File: arrays-hashing/twoSum.py
class Solution:
    def twoSum(self, nums, target):
        #time complexity : O(n)
        #space complexity: O(n)
        numMap = dict()
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in numMap:
                return [numMap[complement], i]
            numMap[nums[i]] = i
        return []

a = Solution()
b = a.twoSum([1, 2, 3, 4, 5], 8)
print(b)

File: javascript/dsa/twoSum.js
var twoSum = function (nums, target) {
    const dict = {};
    for (let j = 0; j < nums.length; j++) {
        let complement = target - nums[j];
        if (complement in dict) {
            return [dict[complement], j];
        }
        dict[nums[j]] = j;
    }
    return [];
};

console.log(twoSum([2, 7, 11, 15], 9));
console.log(twoSum([3, 3], 6));
